src/pcre2_internal.h

```diff
@@ -609,6 +609,34 @@ total length of the tables. */
 #define ctypes_offset (cbits_offset + cbit_length)  /* Character types */
 #define TABLES_LENGTH (ctypes_offset + 256)
 
+/* Private flags used in compile_context.optimization_flags */
+
+#define PCRE2_OPTIM_AUTO_POSSESS    0x00000001u
+#define PCRE2_OPTIM_DOTSTAR_ANCHOR  0x00000002u
+#define PCRE2_OPTIM_START_OPTIMIZE  0x00000004u
+
+/* Tables for converting public optimization directives listed in
+pcre2.h to private optimization flags. */
+
+#define OPTIM_SETBITS \
+  0,                         /* no optimization */   \
+  0xff,                      /* full optimization */ \
```

Comment:
only the least significant byte?

Reply:
to avoid introducing bugs like this one, it would be better to add to pcre2test support to call pcre2_set_optimizations and add some tests.

since we need to keep backward compatibility need to also keep disabling the optimizations by setting the corresponding option flag as well as by calling the new function in the context before compilation.

Reply:
You are absolutely right. Before this PR is marked as ready to merge, I will ensure that the test suite is exercising pcre2_set_optimization().

I wonder if it might not be a good idea to have some unit tests which are written in C, rather than as pcre2test scripts?? Just a thought...